src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpChannelFactory.cs

```diff
@@ -178,10 +178,16 @@ internal HttpClient GetHttpClient()
                         switch (_authenticationScheme)
                         {
                             case AuthenticationSchemes.Basic:
-                                if (credentials.UserName.UserName != string.Empty)
+                                if (credentials.UserName.UserName == null)
                                 {
-                                    creds = new NetworkCredential(credentials.UserName.UserName, credentials.UserName.Password);
+                                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("userName");
                                 }
+                                if (credentials.UserName.UserName == string.Empty)
```

Comment:
Just confirming my understanding of your answer to @dmetzgar -- what does happen on the desktop of you send an empty string or a string containing only whitespace?  And I notice the other AuthenticationSchemes cases below this code are inconsistent both in how they check and what they do if they find string.Empty.  Can you confirm whether we are handling all of these case statements identically to desktop?